### PR TITLE
Hdrp/fix camera with prefab override workflow

### PR DESCRIPTION
**Pull request by RSlysz**
### Purpose of this PR
This fix workflow issue on Camera Inspector:
- Fix prefab override workflows on field was not handled correctly on Camera.
- Fix prefab override workflows on FrameSettings (Though it can only be considered as one big box that can be reverted or apply wholly) 
- Fix FrameSettings multi-edition of overrides (Required cleaning of Serialization API for BitArrays)

Side fix: 
- https://fogbugz.unity3d.com/f/cases/1195508/ (button text alignment issue)
- https://fogbugz.unity3d.com/f/cases/1196598/ (Near alignment issue in Preset)
- https://fogbugz.unity3d.com/f/cases/1196647/ (alignement issue in physical part)

image
Known limitation:
- The FrameSettings can only be applyed or reverted wholly as one entity.
- The blue left bar on the left have small intensity change due to multiple drawing it. This cannot be helped as we need to have the contextual menu on labels but also have only one bar to not confuse user and display it as one prefab overrides.

image

---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: 

---

### CODE REVIEW

**Comment by RSlysz** [comment]
Found a lot of alignment issue in presset that need to be fixed.

File: com.unity.render-pipelines.core/Editor/Utilities/SerializedBitArray.cs [hunk 1]
```diff
@@ -13,18 +45,75 @@ public static bool GetBitArrayAt(this SerializedProperty property, uint bitIndex
...
+                    {
+                        case 8u:    Set8OnOneBitArray(property, bitIndex, value);   return;
+                        case 16u:   Set16OnOneBitArray(property, bitIndex, value);  return;
+                        case 32u:   Set32OnOneBitArray(property, bitIndex, value);  return;
+                        case 64u:   Set64OnOneBitArray(property, bitIndex, value);  return;
+                        case 128u:  Set128OnOneBitArray(property, bitIndex, value); return;
+                        case 256u:  Set256OnOneBitArray(property, bitIndex, value); return;
+                    }
+                }
+                throw new ArgumentException("Trying to call Get on unknown BitArray");
```
Comment (chikuba) [inline com.unity.render-pipelines.core/Editor/Utilities/SerializedBitArray.cs]: s/Get/Set

File: com.unity.render-pipelines.core/Editor/Utilities/SerializedBitArray.cs [hunk 2]
```diff
@@ -13,18 +45,75 @@ public static bool GetBitArrayAt(this SerializedProperty property, uint bitIndex
...
+                        switch (capacity)
+                        {
+                            case 8u:    Set8OnOneBitArray(isolatedProperty, bitIndex, value);   break;
+                            case 16u:   Set16OnOneBitArray(isolatedProperty, bitIndex, value);  break;
+                            case 32u:   Set32OnOneBitArray(isolatedProperty, bitIndex, value);  break;
+                            case 64u:   Set64OnOneBitArray(isolatedProperty, bitIndex, value);  break;
+                            case 128u:  Set128OnOneBitArray(isolatedProperty, bitIndex, value); break;
+                            case 256u:  Set256OnOneBitArray(isolatedProperty, bitIndex, value); break;
+                            default:
+                                throw new ArgumentException("Trying to call Get on unknown BitArray");
```
Comment (chikuba) [inline com.unity.render-pipelines.core/Editor/Utilities/SerializedBitArray.cs]: s/Get/Set

File: com.unity.render-pipelines.core/Editor/Utilities/SerializedBitArray.cs [hunk 3]
```diff
@@ -13,18 +45,75 @@ public static bool GetBitArrayAt(this SerializedProperty property, uint bitIndex
...
+                    }
+                }
+                throw new ArgumentException("Trying to call Get on unknown BitArray");
+            }
+            else
+            {
+                string path = property.propertyPath;
+                foreach (var target in targets)
+                {
+                    // Cannot do better at the moment as bitwise multi eddition only support UInt32 at the moment.
```
Comment (chikuba) [inline com.unity.render-pipelines.core/Editor/Utilities/SerializedBitArray.cs]: Is there a favro card or similar for this? Just so it's not forgotten :)

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs [hunk 4]
```diff
@@ -62,15 +58,38 @@ enum Expandable
 
         readonly static ExpandedState<Expandable, FrameSettings> k_ExpandedState = new ExpandedState<Expandable, FrameSettings>(~(-1), "HDRP");
 
+        static float beginTmp;
```
Comment (chikuba) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs]: This feels a little bit scary, but maybe I'm just not fully understanding why it's done like this?

Comment (RSlysz) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs]: If anyone as better idea to compute the length of the box computed by 
EditorGUILayout.BeginVertical("box");
I will gladly change this.

**Comment by RSlysz** [comment]
Discussed with @fredericv-unity3d This need a little API refactor on the bitarrays

**Pull request merged**